packer-plugin-sdk/acctest/provisioneracc/provisioners.go

```diff
@@ -3,91 +3,252 @@ package provisioneracc
+	dir := fixtureDir()
+	fragmentAbsPath := filepath.Join(dir, templateFragmentPath)
+	fragmentFile, err := os.Open(fragmentAbsPath)
+	if err != nil {
+		return "", fmt.Errorf("Unable find %s", fragmentAbsPath)
+	}
+	defer fragmentFile.Close()
 
-		for vmOS, builderConfig := range builderConfigs {
-			if !provisionerAcc.IsCompatible(builder, vmOS) {
+	fragmentString, err := ioutil.ReadAll(fragmentFile)
+	if err != nil {
+		return "", fmt.Errorf("Unable to read %s", fragmentAbsPath)
+	}
+
+	return string(fragmentString), nil
+}
+
+func RunProvisionerAccTest(testCase *ProvisionerTestCase, t *testing.T) {
+	TestProvisionersAgainstBuilders(testCase, t)
+}
+
+//nolint:errcheck
+func TestProvisionersAgainstBuilders(testCase *ProvisionerTestCase, t *testing.T) {
+	// retrieve user-desired builders.
+	builderTypes := checkBuilders(t)
+
+	// Run this provisioner test case against each builder type requested.
+	for _, builderType := range builderTypes {
+		buildFixtures := BuildersAccTest[builderType]
+		// loop over individual build templates, merge with provisioner
+		// templates, and shell out to run test.
+		for _, buildFixture := range buildFixtures {
+			if !testCase.IsCompatible(builderType, buildFixture.GuestOS) {
 				continue
 			}
 
-			testName := fmt.Sprintf("testing %s builder against %s provisioner", builder, provisioner)
+			testName := fmt.Sprintf("%s on %s", testCase.Name, buildFixture.Name)
+
+			if testCase.Setup != nil {
+				err := testCase.Setup()
+				if err != nil {
+					t.Fatalf("test %s setup failed: %s", testName, err)
+				}
+			}
+
 			t.Run(testName, func(t *testing.T) {
-				provisionerConfig, err := provisionerAcc.GetConfig()
+				builderFragment, err := LoadBuilderFragment(buildFixture.TemplatePath)
 				if err != nil {
-					t.Fatalf("bad: failed to read provisioner config: %s", err.Error())
+					t.Fatalf("failed to load builder fragment: %s", err)
 				}
 
-				// Write json template
+				// Combine provisioner and builder template fragments; write to
+				// file.
 				out := bytes.NewBuffer(nil)
-				fmt.Fprintf(out, `{"builders": [%s],"provisioners": [%s]}`, builderConfig, provisionerConfig)
-				fileName := fmt.Sprintf("%s_%s.json", builder, provisioner)
-				filePath := filepath.Join("./", fileName)
-				writeJsonTemplate(out, filePath, t)
-
-				// set pre-config with necessary builder and provisioner
-				c := buildCommand(t, builderAcc, provisionerAcc)
-				args := []string{
-					filePath,
+				fmt.Fprintf(out, `{"builders": [%s],"provisioners": [%s]}`,
+					builderFragment, testCase.Template)
+				templateName := fmt.Sprintf("%s_%s.json", builderType, testCase.Type)
+				templatePath := filepath.Join("./", templateName)
+				writeJsonTemplate(out, templatePath, t)
+				logfile := fmt.Sprintf("packer_log_%s_%s.txt", builderType, testCase.Type)
+
+				// Run build
+				buildCommand := exec.Command("packer", "build", "--machine-readable", templatePath)
```

Comment:
nice!!

Reply:
If a user does not have Packer installed locally the command will error not so nicely.

```
⇶  ACC_TEST_BUILDERS=amazon-ebs make provisioners-acctest TEST=./provisioner/shell
ACC_TEST_BUILDERS=amazon-ebs go test ./provisioner/shell  -timeout=1h
--- FAIL: TestAccShellProvisioner_basic (0.01s)
    --- FAIL: TestAccShellProvisioner_basic/shell-provisioner-basic_on_Amazon-ebs_Linux_builder (0.01s)
        provisioners.go:237: Error running provisioner acceptance tests: %!s(MISSING)
            Logs can be found at %!s(MISSING) and the acceptance test template can be found at %!s(MISSING)Unable find /home/wilken/Development/pr-reviews/packer/provisioner/shell/test-fixtures/provisioner.shell.txtpacker_log_amazon-ebs_shell.txtamazon-ebs_shell.json
FAIL
FAIL    github.com/hashicorp/packer/provisioner/shell   0.094s
FAIL
make: *** [Makefile:144: provisioners-acctest] Error 1

``` 
We should add an `exec.LookPath("packer")` check here with a detailed error message.

Reply:
good cath!